analyzer/evmverifier/evmverifier.go

```diff
@@ -0,0 +1,183 @@
+type evmContractsVerifier struct {
+	interval time.Duration
+	target   storage.TargetStorage
+	logger   *log.Logger
+	chain    common.ChainName
+	client   *sourcify.SourcifyClient
+}
+
+var _ analyzer.Analyzer = (*evmContractsVerifier)(nil)
+
+func (a *evmContractsVerifier) Name() string {
+	return evmContractsVerifierName
+}
+
+type unverifiedContract struct {
+	Addr                  string
+	AddrContextIdentifier string
+	AddrContextVersion    int
+	AddrData              []byte
+	EthAddress            ethCommon.Address
+}
+
+func (a *evmContractsVerifier) getUnverifiedContracts(ctx context.Context, runtime common.Runtime) ([]*unverifiedContract, error) {
+	var unverifiedContracts []*unverifiedContract
+	rows, err := a.target.Query(ctx, queries.RuntimeEVMUnverfiedContracts, runtime)
+	if err != nil {
+		return nil, fmt.Errorf("querying unverified contracts: %w", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var contract unverifiedContract
+		if err = rows.Scan(
+			&contract.Addr,
+			&contract.AddrContextIdentifier,
+			&contract.AddrContextVersion,
+			&contract.AddrData,
+		); err != nil {
+			return nil, fmt.Errorf("scanning unverified contracts: %w", err)
+		}
+
+		// Compute the eth address from the preimage.
+		ethAddress, err := client.EVMEthAddrFromPreimage(contract.AddrContextIdentifier, contract.AddrContextVersion, contract.AddrData)
+		if err != nil {
+			return nil, fmt.Errorf("contract eth address: %w", err)
+		}
+		// TODO: assert eth address 20 bytes?
+		contract.EthAddress = ethCommon.BytesToAddress(ethAddress)
+
+		unverifiedContracts = append(unverifiedContracts, &contract)
+	}
+	return unverifiedContracts, nil
+}
+
+func (a *evmContractsVerifier) Start(ctx context.Context) {
+	if a.chain != common.ChainNameMainnet && a.chain != common.ChainNameTestnet {
+		a.logger.Warn("EVM contracts verifier only supports testnet/mainnet, stopping", "chain_name", a.chain)
+		return
+	}
+
+	for {
+		select {
+		case <-time.After(a.interval):
+		case <-ctx.Done():
+			a.logger.Warn("shutting down EVM contracts verifier", "reason", ctx.Err())
+			return
+		}
+
+		a.logger.Info("EVM contracts verifier iteration")
+
+		for _, runtime := range evmRuntimes {
+			// Load all non-verified contracts from the DB.
+			unverified, err := a.getUnverifiedContracts(ctx, runtime)
+			if err != nil {
+				a.logger.Error("failed to get unverified contracts", "err", err, "runtime", runtime)
+				continue
+			}
+			if len(unverified) == 0 {
+				a.logger.Debug("no unverified contracts", "runtime", runtime)
+				continue
+			}
+
+			// Query Sourcify for list of all verified contracts.
+			addresses, err := a.client.GetVerifiedContractAddresses(ctx, runtime)
+			if err != nil {
+				a.logger.Error("failed to get verified contract addresses", "err", err, "runtime", runtime)
+				continue
+			}
+			a.logger.Debug("got verified contract addresses", "runtime", runtime, "addresses", addresses)
+			if len(addresses) == 0 {
+				continue
+			}
```

Comment:
nit: does this have any benefits? the rest of the code will run super fast if len==0 here.

Reply:
No benefits atm, but I like having it analogous to the above case when there's no unverified contracts in the db.